common/lib/xmodule/xmodule/modulestore/mongo/draft.py

```diff
@@ -541,6 +544,8 @@ def delete_item(self, location, user_id, revision=None, **kwargs):
             )
         self._delete_subtree(location, as_functions)
 
+        self.do_index(location, delete=True)
```

Comment:
Why are these `do_index()` calls added here? There's no comment and no explanation for what they do, why they're here, and what operations need to call `do_index()` in the future.